python/ray/autoscaler/updater.py

```diff
@@ -181,11 +181,9 @@ def do_sync(remote_path, local_path, allow_non_existing_paths=False):
 
             with LogTimer(self.log_prefix +
                           "Synced {} to {}".format(local_path, remote_path)):
-                if not isinstance(self.cmd_runner, DockerCommandRunner):
-                    # The DockerCommandRunner handles this internally
-                    self.cmd_runner.run(
-                        "mkdir -p {}".format(os.path.dirname(remote_path)),
-                        run_env="host")
+                self.cmd_runner.run(
+                    "mkdir -p {}".format(os.path.dirname(remote_path)),
+                    run_env="host")
```

Comment:
This will break for many Docker uses cases. If I'm trying to sync to the remote location of `/root/file`, running `mkdir -p` on the host will **fail**. Instead, the DockerCommandRunner runs ` mkdir - p {DOCKER_MOUNT_PREFIX}/root/file` on the host, which is fine.

Reply:
This still breaks as we should not special case different command runners. Can you make this internal to DockerCommandRunner? maybe in is init?

Reply:
I think the best solution is to just push this into the `run_rsync_up` codepath.